confgenerator/config.go

```diff
@@ -140,35 +347,118 @@ type LoggingPipeline struct {
 }
 
 // Ops Agent metrics config.
+type metricsReceiverMap map[string]MetricsReceiver
+type metricsProcessorMap map[string]MetricsProcessor
+type metricsExporterMap map[string]MetricsExporter
 type Metrics struct {
-	Receivers  map[string]*MetricsReceiver  `yaml:"receivers"`
-	Processors map[string]*MetricsProcessor `yaml:"processors"`
-	Exporters  map[string]*MetricsExporter  `yaml:"exporters,omitempty"`
-	Service    *MetricsService              `yaml:"service"`
+	Receivers  metricsReceiverMap  `yaml:"receivers" validate:"dive,keys,startsnotwith=lib:"`
+	Processors metricsProcessorMap `yaml:"processors" validate:"dive,keys,startsnotwith=lib:"`
+	Exporters  metricsExporterMap  `yaml:"exporters,omitempty"`
+	Service    *MetricsService     `yaml:"service"`
 }
 
-type MetricsReceiver struct {
-	configComponent `yaml:",inline"`
+type MetricsReceiver interface {
+	component
+}
 
-	CollectionInterval string `yaml:"collection_interval" validate:"required"` // time.Duration format
+type MetricsReceiverShared struct {
+	CollectionInterval string `yaml:"collection_interval" validate:"required,duration=10s"` // time.Duration format
```

Comment:
The value is required to be 10s? Maybe leave a comment to explain.

Reply:
It's required to be at least 10s.